### DISCUSSION TITLE
Multiple records on the same access/inheritence modifiers

### DISCUSSION THREAD

**Discussion by WhiteBlackGoose** [dotnet/csharplang#4898] [Language Ideas] [answered]
Imagine something classic
```cs
public sealed record SomeRecord1;
public sealed record SomeRecord2(int Property1);
public sealed record SomeRecord3;
public sealed record SomeRecord4(string AnotherProperty, int ThirdProperty, bool BlaBla);
```
The idea is to have
```cs
public sealed record
    SomeRecord1,
    SomeRecord2(int Property1),
    SomeRecord3,
    SomeRecord4(string AnotherProperty, int ThirdProperty, bool BlaBla);
```
This makes it much more concise.

What I have not thought about is inheritence. If we are ever to get `closed` on arbitrary types, we would get
```cs
public abstract record MyDU
{
    public sealed record SomeRecord1 : MyDU;
    public sealed record SomeRecord2(int Property1) : MyDU;
    public sealed record SomeRecord3 : MyDU;
    public sealed record SomeRecord4(string AnotherProperty, int ThirdProperty, bool BlaBla) : MyDU;
}
```

Clearly, this `: MyDU` would also be nice to be shortened, but I couldn't think of good syntax for it so far.

**Comment by jnm2** [comment, marked as answer]
Since the C# team is interested in doing something specifically for DUs, we might end up with nice clean syntax for DUs that's even less repetitive than what you're proposing here. Do you have any non-DU examples of where reuse of modifiers would help?

**Reply by WhiteBlackGoose** [reply]
I probably don't. Okay.

**Comment by theunrepentantgeek** [comment]
This would save a handful of characters at the expense of making the code harder to read. 

It also makes it much harder to write any documentation about what the records actually mean, encouraging cryptic code. 

History shows that reducing the number of characters required to write C# is not, of itself, a goal that the LDM usually favours. The introduction of expression bodied members is an exception, purely because the impact is so broad.

**Reply by WhiteBlackGoose** [reply]
Okay, makes sense, thanks
